libraries/api/src/api.js

```diff
@@ -229,7 +229,13 @@ const createRemoteSignatureValidator = (options) => {
     rootUrl: options.rootUrl,
     credentials: {}, // We do this to avoid sending auth headers to authenticateHawk
   });
-  return async (data) => auth.authenticateHawk(data);
+  return async (data, meta) => {
+    let perRequestAuth = auth;
+    if (meta) {
+      perRequestAuth = auth.taskclusterPerRequestInstance(meta);
```

Comment:
As discussed in slack I'll document this functionality. It could be used outside of taskcluster as well if someone really wanted it but we're using it to tell apart special tc libraries from non-special ones and external libraries.